src/components/map/index.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useRef, useEffect, useState } from 'react';
+import mapboxgl from 'mapbox-gl';
+import style from './map.module.css';
+import 'mapbox-gl/dist/mapbox-gl.css';
+
+export type Position = {
+  lat: number;
+  lng: number;
+};
+
+export type MapProps = {
+  initialPosition?: Position;
+};
+
+export function Map({ initialPosition = defaultPosition }: MapProps) {
+  const mapContainer = useRef<HTMLDivElement>(null);
+  const map = useRef<mapboxgl.Map>();
+  const [position, setPosition] = useState<Position>(initialPosition);
+
+  useEffect(() => {
+    if (!mapContainer.current) return;
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      accessToken: process.env.NEXT_PUBLIC_MAPBOX_API_TOKEN,
+      style: process.env.NEXT_PUBLIC_MAPBOX_STOP_PLACES_STYLE_URL,
+      center: [position.lng, position.lat],
+      zoom: 13,
+    });
+
+    return () => map.current?.remove();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (!initialPosition) return;
+    setPosition(initialPosition);
+  }, [initialPosition]);
+
+  useEffect(() => {
+    if (map.current) {
+      map.current.setCenter([position.lng, position.lat]);
+    }
+  }, [position.lat, position.lng]);
```

Comment:
Do we need to store position in separate state here or could we just use position directly in the same way?

Reply:
Hmm. I actually think it is OK to create your own type "boundary" here instead of using common type. I don't think we should propagate geojson position around and tightly couple that type with this API

Reply:
Okay, I'll revert that change then 👍 What is the reason for not doing so? I guess this applies to the Feature type from geojson as well?

Reply:
It's about handling API change. `Position` might not change, but generally, what happens if geojson changes it's API, can we easily map over to be backward compatible or do we have to do an all-out change in the entire codebase? If you use the type directly you are more tightly coupling external dependencies and internal code.